models/userlist.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+//UserList is a list of user.
+// This type provide valuable methods to retrieve information for a group of users efficiently.
+type UserList []*User
+
+//TODO paginate
+
+// IsPublicMember returns true if user public his/her membership in given organization.
+func (ul UserList) IsPublicMember(orgID int64) []bool {
+	results := make([]bool, len(ul))
+	//TODO use directly xorm
+	for i, u := range ul {
+		results[i] = u.IsPublicMember(orgID)
+	}
+	return results
+}
+
+// IsUserOrgOwner returns true if user is in the owner team of given organization.
+func (ul UserList) IsUserOrgOwner(orgID int64) []bool {
+	results := make([]bool, len(ul))
+	//TODO use directly xorm
+	for i, u := range ul {
+		results[i] = u.IsUserOrgOwner(orgID)
+	}
+	return results
+}
+
+// IsTwoFaEnrolled return state of 2FA enrollement
+func (ul UserList) IsTwoFaEnrolled() []bool {
```

Comment:
You should add this to user struct as attribute and replace this method with `loadAttributes` method. See other List structs for example

Reply:
Thx, I am currently going step by step by simply recursively call existing functions, implementing tests and later will optimize to one call.

Reply:
@lafriks Looking at it, I think it would be counter productive as it will need to define a bool prop (that will default to false) at User that we will not fill each time we use directly User object. It would maybe confused futur dev that a prop is not set by default (or falsely set to false).